test/systems/plonk/placeholder/performance.cpp

```diff
@@ -0,0 +1,531 @@
+typename fri_type::params_type create_fri_params(std::size_t degree_log, const int max_step = 1) {
+    typename fri_type::params_type params;
+    math::polynomial<typename FieldType::value_type> q = {0, 0, 1};
+
+    constexpr std::size_t expand_factor = 4;
+
+    std::size_t r = degree_log - 1;
+
+    std::vector<std::shared_ptr<math::evaluation_domain<FieldType>>> domain_set =
+        math::calculate_domain_set<FieldType>(degree_log + expand_factor, r);
+
+    params.r = r;
+    params.D = domain_set;
+    params.max_degree = (1 << degree_log) - 1;
+    params.step_list = generate_random_step_list(r, max_step);
+
+    return params;
+}
+
+// using curve_type = algebra::curves::bls12<381>;
+using curve_type = algebra::curves::pallas;
+using field_type = typename curve_type::base_field_type;
+
+// lpc params
+constexpr static const std::size_t m = 2;
+
+constexpr static const std::size_t table_rows_log = 4;
+constexpr static const std::size_t table_rows = 1 << table_rows_log;
+constexpr static const std::size_t permutation_size = 4;
+constexpr static const std::size_t usable_rows = (1 << table_rows_log) - 3;
+
+struct placeholder_fibonacci_params {
+    using merkle_hash_type = hashes::keccak_1600<512>;
+    using transcript_hash_type = nil::crypto3::hashes::sha2<256>;
+
+    constexpr static const std::size_t witness_columns = 1;
+    constexpr static const std::size_t public_input_columns = 1;
+    constexpr static const std::size_t constant_columns = 0;
+    constexpr static const std::size_t selector_columns = 1;
+
+    using arithmetization_params =
+        plonk_arithmetization_params<witness_columns, public_input_columns, constant_columns, selector_columns>;
+
+    constexpr static const std::size_t lambda = 1;
+    constexpr static const std::size_t m = 2;
+};
+using circuit_fib_params = placeholder_circuit_params<
+    field_type, 
+    typename placeholder_fibonacci_params::arithmetization_params
+>;
+
+constexpr static const std::size_t table_columns =
+    placeholder_fibonacci_params::witness_columns + placeholder_fibonacci_params::public_input_columns;
+
+using transcript_type = typename transcript::fiat_shamir_heuristic_sequential<typename placeholder_fibonacci_params::transcript_hash_type>;
+using lpc_params_type = commitments::list_polynomial_commitment_params<        
+    typename placeholder_fibonacci_params::merkle_hash_type,
+    typename placeholder_fibonacci_params::transcript_hash_type, 
+    placeholder_fibonacci_params::lambda, 
+    placeholder_fibonacci_params::m
+>;
+
+using lpc_type = commitments::list_polynomial_commitment<field_type, lpc_params_type>;
+using lpc_scheme_type = typename commitments::lpc_commitment_scheme<lpc_type>;
+using lpc_placeholder_params_type = nil::crypto3::zk::snark::placeholder_params<circuit_fib_params, lpc_scheme_type>;
+using policy_type = zk::snark::detail::placeholder_policy<field_type, lpc_placeholder_params_type>;
+
+/*
+ACTOR_THREAD_TEST_CASE(small_merkle_tree_test1) {
```

Comment:
Let's remove it?

Reply:
I have refactored this code in the crypto3-zk. I will move them tomorrow, and we will have a good code here.